src/components/evm/AppPage.vue

```diff
@@ -15,19 +15,24 @@ export default defineComponent({
             return Array.isArray(this.tabs);
         },
         selectedTab() {
-            return (this.$route.query.tab ?? this.tabs[0] ?? '') as string;
+            return (this.$route.query.tab ?? '') as string;
         },
     },
     watch: {
-        '$route.query.tab': {
+        $route: {
             immediate: true,
-            handler(newValue) {
-                if (this.tabs === null) {
-                    return;
-                }
+            deep: true,
+            handler(newValue, oldValue = null) {
```

Comment:
Would it be more intuitive to use a default value for oldValue directly in the method parameters, like handler(newValue, oldValue = {}) {
then you can remove null checks?

Reply:
in this case the comparison on the next line `oldValue === null` needs to be there because on initial page load, `oldValue` is undefined. if i make it an empty object, i can no longer do the comparison because `oldValue === {}` will always be false, since js compares objects by reference, not value.